src/Atc.Rest.Client/Builder/IMessageRequestBuilder.cs

```diff
@@ -39,27 +39,37 @@ IMessageRequestBuilder WithPathParameter<T>(string name, T value)
             where T : struct;
 
         /// <summary>
-        /// Adds a query parameter to the created request URL.
+        /// Adds a value to a header parameter in the headers.
         /// </summary>
-        /// <remarks>
-        /// If the <paramref name="value"/> is null, the query parameter is not added.
-        /// </remarks>
-        /// <param name="name">Name of the query parameter.</param>
-        /// <param name="value">Value of the query parameter.</param>
+        /// <param name="name">Name of the header parameter.</param>
+        /// <param name="value">Value to use as the header parameter.</param>
+        /// <exception cref="ArgumentException">Thrown when <paramref name="value"/> is null, empty, or whitespace.</exception>
         /// <returns>The <see cref="IMessageRequestBuilder"/>.</returns>
-        IMessageRequestBuilder WithQueryParameter(string name, string? value);
+        IMessageRequestBuilder WithHeaderParameter(string name, string value);
```

Comment:
Before the who repo problems the other day, I had a PR with this feature in it. It did however use a strongly typed approach, e.g.:

```c#
AuthenticationHeaderValue headerValue = ...;
sut.WithHeader(x => x.Authorization, headerValue);
```

It makes it much harder to misspell a header name, or pass in an invalid value, since the parameter, `AuthenticationHeaderValue ` in the example above, validates the value assigned to it.

I have code and tests ready if you like me to reintroduce it.